src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/HomeOfficeCaseNotificationsHandler.java

```diff
@@ -1,169 +1,218 @@
+
+@CacheConfig(cacheNames = {"caseId","event"})
+@Component
+@Slf4j
+public class HomeOfficeCaseNotificationsHandler implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private static final String SUPPRESSION_LOG_FIELDS = "event: {}, "
+                                                         + "caseId: {}, "
+                                                         + "homeOfficeReferenceNumber: {}, "
+                                                         + "homeOfficeSearchStatus: {}, "
+                                                         + "homeOfficeNotificationsEligible: {} ";
+    private final FeatureToggler featureToggler;
+    private final HomeOfficeApi<AsylumCase> homeOfficeApi;
+    private Long cacheCaseId = 99999L;
+    private String cachedEvent;
+
+    private static final String HO_NOTIFICATION_FEATURE = "home-office-notification-feature";
+
+    @Override
+    public DispatchPriority getDispatchPriority() {
+        return DispatchPriority.LATEST;
+    }
+
+    public HomeOfficeCaseNotificationsHandler(
+        FeatureToggler featureToggler,
+        HomeOfficeApi<AsylumCase> homeOfficeApi) {
+        this.featureToggler = featureToggler;
+        this.homeOfficeApi = homeOfficeApi;
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        AsylumCase asylumCaseWithHomeOfficeData =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        AppealType appealType = asylumCaseWithHomeOfficeData.read(APPEAL_TYPE, AppealType.class)
+                .orElseThrow(() -> new IllegalStateException("AppealType is not present."));
+
+        if (!HomeOfficeAppealTypeChecker.isAppealTypeEnabled(featureToggler, appealType)) {
+
+            return new PreSubmitCallbackResponse<>(asylumCaseWithHomeOfficeData);
+        }
+
+        final String homeOfficeSearchStatus = asylumCaseWithHomeOfficeData.read(HOME_OFFICE_SEARCH_STATUS, String.class)
+            .orElse("");
+        final YesOrNo homeOfficeNotificationsEligible
+            = asylumCaseWithHomeOfficeData.read(HOME_OFFICE_NOTIFICATIONS_ELIGIBLE, YesOrNo.class)
+            .orElse(YesOrNo.NO);
+        final long caseId = callback.getCaseDetails().getId();
+        final String homeOfficeReferenceNumber
+            = asylumCaseWithHomeOfficeData.read(HOME_OFFICE_REFERENCE_NUMBER, String.class).orElse("");
+
+        if (asylumCaseWithHomeOfficeData.read(APPELLANT_IN_UK, YesOrNo.class).map(
+            value -> value.equals(YesOrNo.YES)).orElse(true)) {
+
+            // RIA-5683: Prevent multiple notifications for the same event and case ID.
+            if (Objects.equals(cachedEvent, callback.getEvent().toString()) && cacheCaseId == caseId) {

```

Comment:
Maybe EqualsIgnoreCase just in case that for some reason the send a lower case event in the future?